GetEmissionHistory.cs

```diff
@@ -0,0 +1,173 @@
+        };
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.BadRequest, responseBodyObject);
+      } catch (Exception) {
+        return await HttpResponseDataFactory.GetHttpResponseData(req, HttpStatusCode.InternalServerError, responseBodyObject);
+      }
+      throw new NotImplementedException();
+    }
+
+    private int GetUserIdIfUserKeyExistsInDB(string userKey) {
+      MySqlConnection connection = DatabaseConnecter.MySQLDatabase();
+
+      using (connection)
+      {
+        connection.Open();
+
+        try
+        {
+          using (MySqlCommand command = connection.CreateCommand())
+          {
+            string query = "SELECT ID, UserKey FROM Users";
+            command.CommandText = query;
+            using MySqlDataReader reader = command.ExecuteReader();
+            if (reader.HasRows) {
+              while (reader.Read())
+              {
+                string hashedUserKeyInDB = reader.GetString(1);
+                if (BCrypt.Net.BCrypt.Verify(userKey, hashedUserKeyInDB)) {
+                  connection.Close();
+                  return reader.GetInt32("ID");
+                }
+              }
+            }
+          }
+          connection.Close();
+          return -1;
+        } catch (Exception) {
+          connection.Close();
+          return -1;
+        }
+      }
+    }
+
+    private async Task<List<DailyEmission>> GetDailyEmissionsForUserWithinOneYear(int userId)
+    {
+      DateTime currentDate = DateTime.UtcNow;
+      DateTime oneYearAgo = currentDate.AddYears(-1);
+
+      MySqlConnection connection = DatabaseConnecter.MySQLDatabase();
+
+      using (connection)
+      {
+        connection.Open();
+
+        string query = "SELECT e.DateTime, e.TotalAmount, e.Goal " +
+               "FROM DailyEmission AS e " +
+               "JOIN Users AS u ON e.UserID = u.ID " +
+               "WHERE u.ID = @userId AND e.DateTime >= '@oneYearAgo' AND e.DateTime <= '@currentDate'" +
+               " ORDER BY e.DateTime ASC";
+
+
+        using (MySqlCommand command = connection.CreateCommand())
+        {
+          command.CommandText = query;
+          command.Parameters.AddWithValue("@userId", userId);
+          command.Parameters.AddWithValue("@oneYearAgo", oneYearAgo.ToString("yyyy/MM/dd"));
+          command.Parameters.AddWithValue("@currentDate", currentDate.ToString("yyyy/MM/dd"));
+
+          using (MySqlDataReader reader = await command.ExecuteReaderAsync())
+          {
+            List<DailyEmission> emissions = new List<DailyEmission>();
+            const string DATE_TIME_COL = "DateTime";
+            const string TOTAL_AMOUNT_COL = "TotalAmount";
+            const string GOAL_COL = "Goal";
+
+            if (reader.HasRows)
+            {
+              while (reader.Read())
+              {
+                // Store a list of DailyEmission
+                DateTime dateTime = reader.GetDateTime(DATE_TIME_COL).Date;
```

Comment:
I think it would make more sense to keep it as a string for the return, since when you retrieve the date in line 124 & 125 its a string. This brings to the greater point, i think we should change the database specification to string instead of date.

Reply:
I think we should keep it as date datatype. it is fine as long as you pass C# DateTime object into `command.Parameters.AddWithValue("@DateTime", date)`.

By passing DateTime object instead of string has the benefits of faster performance (condition handling) in date comparison in SQL query. If I use a string instead, for every entry in the query search, I have to use some SQL function like DATE() to convert the string into DateTime and compare the string like "yyyy/mm/dd" in sql query.

However, there are things that I missed: I forgot to format the dateTime result above into a string format "yyyy/mm/dd" for returning in the function.